nodeJS/express/introduction_to_express.md

```diff
@@ -0,0 +1,98 @@
+
+Once you navigate to `http://localhost:3000/`, you should see `Hello, world!` appear in the window. Let's break down how our Express code got us here.
+
+When our server receives our `GET` request, Express stores the request in a [`request` object](https://expressjs.com/en/4x/api.html#req). This request gets passed through a chain of functions we call `middleware functions` until eventually, a middleware function tells Express to actually respond to the request.
+
+In our example, the request comes through as a `GET` request to the `/` route. This matches the route handler we have.
+
+```javascript
+app.get("/", (req, res) => res.send("Hello, world!"));
+```
+
+We will break down routes and route handlers in more detail in a later lesson, but to summarise the above line, it tells Express "if a `GET` request comes through to the `/` route, run the request through the following chain of middleware functions". Here, we only have a single middleware function in this chain.
+
+If we had defined multiple route handlers, Express would pass the request through the first one to match the requested route. Order matters!
+
+Express takes the callback function we gave it and passes the request object into the first parameter (commonly named `req` by convention), and a [`response` object](https://expressjs.com/en/4x/api.html#res) into the second parameter (commonly named `res`). Our callback tells the response object to respond to the request by `.send`ing the string `"Hello, world!"`.
+
+There is no more code to run and the function returns. Since Express has been told to respond to the request, it ends the request here. Meanwhile, the browser receives our server's response and displays it on screen, which is our `"Hello, world!"` string. We could send nearly anything in response. We could even [tell Express to send a file](https://expressjs.com/en/api.html#res.sendFile).
+
+### Nodemon
+
+After you run your server with `node app.js`, if you make any changes to any JavaScript and JSON files in the project directory, these changes will not be reflected in the server unless you manually interrupt and rerun `node app.js`. Similarly to how Webpack and Vite's dev servers would restart and rebuild your site when you make a change to a file, you can use [Nodemon](https://www.npmjs.com/package//nodemon) to run your `app.js` file.
+
+Our recommendation would be to add it as a dev dependency and write an npm script that runs `nodemon app.js`.
+
+### Assignment
+
+<div class="lesson-content__panel" markdown="1">
+
+1. Go back to your [Basic Informational Site project](https://www.theodinproject.com/lessons/nodejs-basic-informational-site), install Express and rewrite the project using it! You should be able to do most of this with just a few `app.get()`s.
```

Comment:
We can add some reading resources from the Express docs. This will be super valuable for our learners.

This are the sections I have in mind (as they are already covered in the lesson):

- https://expressjs.com/en/4x/api.html#app.get.method
- https://expressjs.com/en/4x/api.html#app.listen
- https://expressjs.com/en/4x/api.html#res.send

Reply:
I was of the mind that at least for app.get and res.send, they'd be more suitable in the routes lesson perhaps? Since that's when we'd ideally be breaking down the anatomy of a route handler and middleware function, where both of these can be discussed in more detail.

And I feel app.listen as a sole "extra reading" feels a bit off, though that may just be me.

Reply:
Rather than specific links, we can just link the Express site https://expressjs.com/ and ask learners to explore the documentation on their own with wording similar to "Spend a few minutes exploring the Express documentation so you can get a feel for it. We'll be referencing a lot of content from the docs in the next lessons"

Reply:
Sensible - I'll add that as an assignment step before the basic info site rewrite.